src/status_im/transport/inbox.cljs

```diff
@@ -0,0 +1,191 @@
+(ns status-im.transport.inbox
+  (:require [re-frame.core :as re-frame]
+            [status-im.native-module.core :as status]
+            [status-im.utils.handlers :as handlers]
+            [status-im.transport.utils :as web3.utils]
+            [taoensso.timbre :as log]))
+
+(defn- parse-json
+  ;; NOTE(dmitryn) Expects JSON response like:
+  ;; {"error": "msg"} or {"result": true}
+  [s]
+  (try
+    (let [res (-> s
+                  js/JSON.parse
+                  (js->clj :keywordize-keys true))]
+      ;; NOTE(dmitryn): AddPeer() may return {"error": ""}
+      ;; assuming empty error is a success response
+      ;; by transforming {"error": ""} to {:result true}
+      (if (and (:error res)
+               (= (:error res) ""))
+        {:result true}
+        res))
+    (catch :default e
+      {:error (.-message e)})))
+
+(defn- response-handler [error-fn success-fn]
+  (fn handle-response
+    ([response]
+     (let [{:keys [error result]} (parse-json response)]
+       (handle-response error result)))
+    ([error result]
+     (if error
+       (error-fn error)
+       (success-fn result)))))
+
+(defn add-peer [enode success-fn error-fn]
+  (status/add-peer enode (response-handler error-fn success-fn)))
+
+(defn fetch-peers
+  ;; https://github.com/ethereum/go-ethereum/wiki/Management-APIs#admin_peers
+  ;; retrieves all the information known about the connected remote nodes
+  ;; TODO(dmitryn): use web3 instead of rpc call
+  [success-fn error-fn]
+  (let [args    {:jsonrpc "2.0"
+                 :id      2
+                 :method  "admin_peers"
+                 :params  []}
+        payload (.stringify js/JSON (clj->js args))]
+    (status/call-web3 payload (response-handler error-fn success-fn))))
+
+(defn registered-peer? [peers enode]
+  (let [peer-ids (into #{} (map :id) peers)
+        enode-id (web3.utils/extract-enode-id enode)]
+    (contains? peer-ids enode-id)))
+
+(defn mark-trusted-peer [web3 enode peers success-fn error-fn]
+  (.markTrustedPeer (web3.utils/shh web3)
+                    enode
+                    (fn [err resp]
+                      (if-not err
+                        (success-fn resp)
+                        (error-fn err)))))
+
+(defn request-messages [web3 wnode topics to from sym-key-id success-fn error-fn]
```

Comment:
Docstring, worth noting this doesn't actually work sync which is what name would have you believe. Instead it is more like a "poke", and then messages appear in filter. (The fact that this wasn't documented, something I should've done, meant that several people spent in total a few/several hours being confused by this).